src/ui/widgets/Symbol/symbol.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+
+import { Widget } from "../widget";
+import { PVWidgetPropType, PVComponent } from "../widgetProps";
+import {
+  InferWidgetProps,
+  BoolPropOpt,
+  StringPropOpt,
+  ColorPropOpt,
+  BoolProp,
+  FloatPropOpt,
+  BorderPropOpt,
+  StringProp
+} from "../propTypes";
+import { registerWidget } from "../register";
+import { LabelComponent } from "../Label/label";
+import { Color } from "../../../types/color";
+import { ImageComponent } from "../Image/image";
+
+const SymbolProps = {
+  src: StringProp,
+  alt: StringPropOpt,
+  backgroundColor: ColorPropOpt,
+  showLabel: BoolProp,
+  width: FloatPropOpt,
+  height: FloatPropOpt,
+  border: BorderPropOpt,
+  value: FloatPropOpt,
+  rotation: FloatPropOpt,
+  flipHorizontal: BoolPropOpt,
+  flipVertical: BoolPropOpt,
+  visible: BoolPropOpt,
+  stretchToFit: BoolPropOpt
+};
+
+export type SymbolComponentProps = InferWidgetProps<typeof SymbolProps> &
+  PVComponent;
+
+/**
+ * This component combines the use of a svg with a label, and is used to replace
+ * the MultistateMonitorWidget from CS-Studio
+ * @param props
+ */
+export const SymbolComponent = (props: SymbolComponentProps): JSX.Element => {
+  const { showLabel, visible = true } = props;
+  const background = props.backgroundColor || Color.WHITE;
+
+  const useSizeProperties = !props.showLabel || props.stretchToFit;
+  const sizeProps = {
+    width: useSizeProperties ? `${props.width}px` : undefined,
+    height: useSizeProperties ? `${props.height}px` : undefined
+  };
+
+  return (
+    <div
+      style={{
+        backgroundColor: "transparent"
+      }}
+    >
+      <ImageComponent
+        {...{
+          ...props,
+          ...sizeProps
+        }}
+      />
+      {showLabel && (
+        <LabelComponent
+          {...{ visible, backgroundColor: background, ...props }}
+          text={props.value?.getStringValue() ?? ""}
+          // TODO: This is pretty hacky, on CS-Studio when using stretchToFit
+          // the text stays still while the image expands underneath, to compensate
+          // for expansion of the image the text is shifted a FIXED amount
+          transform={`translate(0px, ${props.stretchToFit ? -20 : 0}px)`}
```

Comment:
@willrogers here is the hacky code, I know it is bad but to do it properly requires the size of the original image to determine the effect stretchToFit will have on the text location. Do we care enough about this specific feature to implement it properly? If there is an easy method I am missing then let me know, but otherwise from the amount stretchToFit is used in CS-Studio I would assume not.

Reply:
I'm happy to merge this now, because I think it's a step forward anyway, but I agree that this looks fishy. Can you put an example on the widget test page that demonstrates the problem you're solving?

[Another example where just being able to look at your PC would probably save some time.]

Reply:
I can place a screenshot here so it remains somewhere accessible.
Effectively the first part of the image is with stretchToFit false, the image is the right size and the text is just below it.
The second part of the image is with stretchToFit true, which causes the image to expand (like it does in CS-Studio), but means that the text is pushed further down because it is placed after the image.
Third is the desired behaviour from CS-Studio, where the image is stretched but the text stays where it was before, this is why this line is added, as I cannot think of a proper way of doing it.
![StretchToFit](https://user-images.githubusercontent.com/27682458/97457959-27b27c80-1932-11eb-9bad-2008622120ea.png)

Reply:
OK, thanks, I suggest we merge and move on.